### PR TITLE
Dehardcode Rental Expire/Renting item checks

### PR DESCRIPTION
**Pull request by hemagx**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Implemented new script fields in item db executes when item is rented and when item rent expires
Implemented a new script command getfont() gets the current player font id
Fixed hardcoded checks for rental items in source

**Issues addressed:** 
#140

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2462)

**Comment by 4144** [comment]
what will happened if other script already running and rental script will be run?

File: src/map/pc.c [hunk 1]
```diff
@@ -4820,12 +4758,21 @@ static int pc_delitem(struct map_session_data *sd, int n, int amount, int type,
 
 	sd->status.inventory[n].amount -= amount;
 	sd->weight -= sd->inventory_data[n]->weight*amount ;
+
+	// It's here because the data would most likely get zeroed in following if [Hemagx]
+	struct item_data *itd = sd->inventory_data[n];
+	bool is_rentral = (sd->status.inventory[n].expire_time > 0) ? true : false;
```
Comment (dastgirp) [inline src/map/pc.c]:
```suggestion
	bool is_rental = (sd->status.inventory[n].expire_time > 0) ? true : false;
```

File: src/map/script.c [hunk 2]
```diff
@@ -25454,6 +25467,36 @@ static void script_run_item_unequip_script(struct map_session_data *sd, struct i
 	script->current_item_id = 0;
 }
 
+static void script_run_item_rental_start_script(struct map_session_data *sd, struct item_data *data, int oid) __attribute__((nonnull(1, 2)));
```
Comment (dastgirp) [inline src/map/script.c]: Why is there declaration of function here?

**Comment by hemagx** [comment]
@4144 Hmm, good point. we can probably set an ```sd->npc_id```(Not sure if the variable name correct), check, since rentals on regular basis checked a few seconds extra of renting is not a problem.

File: src/map/script.c (hunk 2 above)
Comment (hemagx) [reply src/map/script.c]: it has a nonull attribute

**Comment by 4144** [comment]
i mean if you do attach to player, it will terminate previous script. but i not sure is you attaching script to player or run in deattached

**Comment by hemagx** [comment]
@4144 Item scripts works in attached mode, yet it can still be used while talking/executing NPC perfectly fine.

**Pull request merged**